src/de/decoder.rs

```diff
@@ -16,41 +16,51 @@ use crate::{config::Config, error::DecodeError, utils::Sealed};
 /// # let slice: &[u8] = &[0, 0, 0, 0];
 /// # let some_reader = bincode::de::read::SliceReader::new(slice);
 /// use bincode::de::{DecoderImpl, Decode};
-/// let mut decoder = DecoderImpl::new(some_reader, bincode::config::standard());
+/// let mut context = ();
+/// let mut decoder = DecoderImpl::new(some_reader, bincode::config::standard(), &mut context);
 /// // this u32 can be any Decode
 /// let value = u32::decode(&mut decoder).unwrap();
 /// ```
-pub struct DecoderImpl<R, C: Config> {
+pub struct DecoderImpl<'context, R, C: Config, Context> {
     reader: R,
     config: C,
     bytes_read: usize,
+    context: &'context mut Context,
 }
 
-impl<R: Reader, C: Config> DecoderImpl<R, C> {
+impl<'context, R: Reader, C: Config, Context> DecoderImpl<'context, R, C, Context> {
     /// Construct a new Decoder
-    pub const fn new(reader: R, config: C) -> DecoderImpl<R, C> {
+    pub fn new(
+        reader: R,
+        config: C,
+        context: &'context mut Context,
+    ) -> DecoderImpl<'context, R, C, Context> {
         DecoderImpl {
             reader,
             config,
             bytes_read: 0,
+            context,
         }
     }
 }
 
-impl<R, C: Config> Sealed for DecoderImpl<R, C> {}
+impl<R, C: Config, Context> Sealed for DecoderImpl<'_, R, C, Context> {}
 
-impl<'de, R: BorrowReader<'de>, C: Config> BorrowDecoder<'de> for DecoderImpl<R, C> {
+impl<'de, R: BorrowReader<'de>, C: Config, Context> BorrowDecoder<'de>
```

Comment:
I wonder if we can make the `Context: ?Sized`

Reply:
No because the DecoderImpl owns a value of `Context`. Making it unsized would make weird problems. Unless we made it references all the way up.

Reply:
Nvm, I'm dumb. I was looking at a local change that removed that :sigh: